include/deal.II/base/mpi.h

```diff
@@ -704,6 +789,42 @@ namespace Utilities
        * MPI process.
        */
       ~MPI_InitFinalize();
+
+      /**
+       * Register a reference to an MPI_Request
+       * that we MPI_Wait on before calling MPI_Finalize.
+       *
+       * The object @p request needs to exist when MPI_Finalize is called, which means the
+       * request is typically statically allocated. Note that is is acceptable
+       * for a request to be already waited on (and consequently reset to
+       * MPI_REQUEST_NULL).
+       *
+       * A typical usage is the following idiom:
+       * @code
+       * void my_fancy_communication()
+       * {
+       *   static MPI_Request request = MPI_REQUEST_NULL;
+       *   MPI_InitFinalize::register_static_request(request);
+       *   MPI_Wait(&request, MPI_STATUS_IGNORE);
+       *   // [some algorithm that is not safe to be executed twice in a row.]
+       *   MPI_IBarrier(comm, &request);
```

Comment:
The example doesn't show the use of `unregister_request()`.

It would also be good to explicitly state that calling the function twice on the same request object is ok.